python_modules/dagstermill/dagstermill/cli.py

```diff
@@ -0,0 +1,93 @@
+import os
+import sys
+import click        
+import re
+
+from dagster.cli.dynamic_loader import repository_target_argument, load_target_info_from_cli_args, get_module_target_function
+
+def get_scaffolding(register_repo_str):
+    STARTING_NOTEBOOK_INIT = '''
+        {{
+        "cells": [
+        {{
+        "cell_type": "code",
+        "execution_count": null,
+        "metadata": {{}},
+        "outputs": [],
+        "source": [
+            "import dagstermill as dm\\n",
+            "{0}\\n",
+            "{1}"
+        ]
+        }},
+        {{
+        "cell_type": "code",
+        "execution_count": null,
+        "metadata": {{
+            "tags": [
+            "parameters"
+            ]
+        }},
+        "outputs": [],
+        "source": [
+            "dm_context = dm.define_context()"
+        ]
+        }}
+        ],
+        "metadata": {{
+        "celltoolbar": "Tags"
+        }},
+        "nbformat": 4,
+        "nbformat_minor": 2
+        }}
+    '''
+    return STARTING_NOTEBOOK_INIT.format(register_repo_str[0], register_repo_str[1])
+
+def create_dagstermill_cli():
+    return ui
+
+@click.command(
+    name='ui',
+    help=(
+        'Creates new dagstermill notebook.'
+    ),
+)
+@repository_target_argument
+@click.option('--notebook', '-note', type=click.STRING, help="Name of notebook")
+@click.option('--solid-name', '-s', default="", type=click.STRING, help="Name of solid that represents notebook in the repository. If empty, defaults to notebook name.")
+@click.option('--jupyter', '-j', is_flag = True, help="Launches jupyter notebook and opens to newly created notebook")
+@click.option('--force-overwrite', is_flag = True, help="Will force overwrite any existing notebook or file with the same name.")
+def ui(notebook, solid_name, jupyter, force_overwrite, **kwargs):
+    
+    #@Uma TODO: we might want to sanitize the notebook name so that it's a valid filename
+    #Perhaps something like this: https://github.com/django/django/blob/master/django/utils/text.py function: get_valid_filename()
+
+    if not notebook.endswith(".ipynb"):
+        notebook += ".ipynb"
+    notebook_path = os.path.join(os.getcwd(), notebook)
+
+    if not force_overwrite and os.path.isfile(notebook_path):
+        click.confirm("Warning, {notebook_path} already exists and continuing will overwrite the existing notebook. Are you sure you want to continue?".format(notebook_path=notebook_path), abort=True)
+
+    if solid_name == "":
+        solid_name = os.path.basename(notebook_path).split(".")[0]
```

Comment:
in this case that's cool because you are assigning a default value and not changing the meaning of the variable